File: sample/src/main/java/vn/tiki/sample/login/LoginView.java
package vn.tiki.sample.login;

import vn.tiki.architecture.mvp.Mvp;

/**
 * Created by Giang Nguyen on 8/25/17.
 */

public interface LoginView extends Mvp.View {

  void showLoading();

  void hideLoading();

  void showValidationEmailError();

  void hideValidationEmailError();

  void showValidationPasswordError();

  void hideValidationPasswordError();

  void enableSubmit();

  void disableSubmit();

  void showAuthenticationError();

  void hideAuthenticationError();

  void showNetworkError();

  void hideNetworkError();

  void showLoginSuccess();
}

File: sample/src/main/java/vn/tiki/sample/login/LoginView.kt
package vn.tiki.sample.login

import vn.tiki.architecture.mvp.Mvp

interface LoginView : Mvp.View {

  fun showLoading()

  fun hideLoading()

  fun showValidationEmailError()

  fun hideValidationEmailError()

  fun showValidationPasswordError()

  fun hideValidationPasswordError()

  fun enableSubmit()

  fun disableSubmit()

  fun showAuthenticationError()

  fun hideAuthenticationError()

  fun showLoginSuccess()
}
